backend/x86_proc.ml

```diff
@@ -15,6 +15,70 @@
 
 open X86_ast
 
+module StringTbl = Hashtbl.Make (struct
+    type t = string
+
+    let equal = String.equal
+
+    let hash = Hashtbl.hash
+  end)
+
+module SectionName = struct
+  type t =
+    { name: string list;
+      name_str: string;
+      flags : string option;
+      args: string list
+    }
+
+  let equal t1 t2 =
+    List.equal (String.equal) t1.name t2.name
+
+  let hash t = Hashtbl.hash t.name
+
+  let compare t1 t2 = List.compare (String.compare) t1.name t2.name
+
+  let make name flags args =
+    { name; name_str = String.concat "," name; flags; args; }
+
+  let from_name name = { name = [name]; name_str = name; flags = None; args = [] }
+
+  let name t = t.name_str
+
+  let flags t =
+    let parse_flags flags =
+      let rec inner acc = function
+        | Seq.Nil -> acc
+        | Seq.Cons (c, tl) ->
+          let flag =
+            match c with
+            (* CR mcollins - modify types to avoid string comparisons *)
+            | 'a' -> 0x2L
+            | 'w' -> 0x1L
+            | 'x' -> 0x4L
+            | 'M' -> 0x10L
+            | 'S' -> 0x20L
+            | '?' -> 0x0L
+            | 'G' -> 0x0L
+            | _ ->
+              failwith (Printf.sprintf "Unknown flag %c in flags %s\n" c flags)
+          in
+          inner (Int64.logor acc flag) (tl ())
+      in
+      inner 0L (String.to_seq flags ()) in
+    Option.fold ~none:0L ~some:parse_flags t.flags
+
+  let alignment t =
+    let rec align = function
+      | [] -> 0L
+      | [hd] -> Option.value ~default:0L (Int64.of_string_opt hd)
```

Comment:
Is the alignment guarangeed to be the last arg? A comment here would be helpful.

Reply:
I have no idea, thats just what the compiler is currently generating

Reply:
Yeah, this seems to be as existing.  There are various things in the DSL that could probably be cleaned up later.